File: 프로그래머스/1/81301. 숫자 문자열과 영단어/숫자 문자열과 영단어.js
function solution(s) {
    compareNum.map((i)=>
      s= s.replaceAll(i.english,i.number));
    return parseInt(s);
}

const compareNum= [
    {number:0,english:'zero'},
    {number:1,english:'one'},
    {number:2,english:'two'},
    {number:3,english:'three'},
    {number:4,english:'four'},
    {number:5,english:'five'},
    {number:6,english:'six'},
    {number:7,english:'seven'},
    {number:8,english:'eight'},
    {number:9,english:'nine'}
]

File: 프로그래머스/1/81301. 숫자 문자열과 영단어/숫자 문자열과 영단어.java
import java.util.HashMap;
import java.util.Map;

public class Solution {
    public static int solution(String s) {
        Map<String, Integer> compareNum = new HashMap<>();
        compareNum.put("zero", 0);
        compareNum.put("one", 1);
        compareNum.put("two", 2);
        compareNum.put("three", 3);
        compareNum.put("four", 4);
        compareNum.put("five", 5);
        compareNum.put("six", 6);
        compareNum.put("seven", 7);
        compareNum.put("eight", 8);
        compareNum.put("nine", 9);

        for (Map.Entry<String, Integer> entry : compareNum.entrySet()) {
            s = s.replaceAll(entry.getKey(), entry.getValue().toString());
        }

        return Integer.parseInt(s);
    }

    public static void main(String[] args) {
        System.out.println(solution("one4seveneight")); // 예시 입력
    }
}
